src/generator/Generator.ts

```diff
@@ -111,10 +131,55 @@ class Generator {
         }
     }
 
-    private visitBlockStatement(state: VisitorState) {
-        state.body = this.module.block('', state.statements);
+    private visitIfStatement(node: IfStatement, parent: TraversalAncestors, state: VisitorState) {
+        const condition = state.expressionStack.pop();
+
+        if (condition === undefined) {
+            throw new Error('Undefined condition in if statement');
+        }
+
+        const branches = state.branches.get(node);
+
+        if (branches === undefined) {
+            throw new Error('Missing branch statement');
+        }
+
+        const [ifPart, elsePart] = branches;
+        const ifStatement = this.module.if(condition, ifPart, elsePart);
+        const immediateParent = Generator.getImmediateParent(parent);
+
+        if (immediateParent !== undefined && isIfStatement(immediateParent)) {
+            const parentBranches = state.branches.get(immediateParent);
+
+            if (parentBranches === undefined) {
+                throw Error('Parent branches are undefined');
+            }
+
+            parentBranches[1] = ifStatement;
+            state.branches.set(immediateParent, parentBranches);
```

Comment:
Is this really necessary? You only change the value in an array. Therefore the reference should not have changed and the set call can savely be deleted.